zcash_client_sqlite/src/lib.rs

```diff
@@ -115,9 +126,18 @@ impl<P: consensus::Parameters> WalletDB<P> {
                 stmt_select_tx_ref: self.conn.prepare(
                     "SELECT id_tx FROM transactions WHERE txid = ?",
                 )?,
-                stmt_mark_recived_note_spent: self.conn.prepare(
+                stmt_mark_sapling_note_spent: self.conn.prepare(
                     "UPDATE received_notes SET spent = ? WHERE nf = ?"
                 )?,
+                stmt_mark_transparent_utxo_spent: self.conn.prepare(
+                    "UPDATE utxos SET spent_in_tx = :spent_in_tx
+                    WHERE prevout_txid = :prevout_txid
+                    AND prevout_idx = :prevout_idx"
+                )?,
+                stmt_insert_received_transparent_utxo: self.conn.prepare(
+                    "INSERT INTO utxos (address, prevout_txid, prevout_idx, script, value_zat, height)
```

Comment:
Will `utxoid` get set automatically with this insert statement? Even though it's not listed as "autoincrement" in the [create statement](https://github.com/zcash/librustzcash/pull/341/files#diff-54ec5e42408ac28d55641fbe89818c7a8ea21c666b6e4b79b059776661c4bce9R111)?

I'm guessing that rusqlite handles that, by default?

Reply:
The main thing we want to make sure of is that calling insert with the same UTXO, twice will not result in multiple rows for this table.

Reply:
Yup, the `PRIMARY KEY` declaration covers that.